lib/uninterleaveAndPackBuffers.js

```diff
@@ -24,148 +24,122 @@ module.exports = uninterleaveAndPackBuffers;
  * @see loadGltfUris
  */
 function uninterleaveAndPackBuffers(gltf) {
-    var accessors = gltf.accessors;
-    var buffers = gltf.buffers;
-    var packBufferViews = {length : 0};
-    for (var bufferId in buffers) {
-        if (buffers.hasOwnProperty(bufferId)) {
-            packGltfBuffer(gltf, bufferId, packBufferViews);
-        }
-    }
-    delete packBufferViews.length;
-    gltf.bufferViews = packBufferViews;
+    var arrayBufferDataOffset = 0;
+    var arrayBufferDataLength = 0;
+    var elementArrayBufferDataOffset = 0;
+    var elementArrayBufferDataLength = 0;
+    var otherDataOffset = 0;
+    var otherDataLength = 0;
 
-    // Change the accessor bufferViews to point to the new bufferViews
-    for (var accessorId in accessors) {
-        if (accessors.hasOwnProperty(accessorId)) {
-            var accessor = accessors[accessorId];
-            accessor.bufferView = accessor.extras._pipeline.bufferView;
+    // compute the total size for each arraybuffer type
+    ForEach.accessor(gltf, function (accessor) {
+        var bufferView = gltf.bufferViews[accessor.bufferView];
```

Comment:
@lasalvavida I think in the most recent spec. Accessor doesn't have to have a bufferView property?